lib/plugins/aws/lib/naming.test.js

```diff
@@ -549,9 +549,9 @@ describe('#naming()', () => {
   });
 
   describe('#getCloudWatchLogLogicalId()', () => {
-    it('should normalize the function name and add the standard suffix including the index', () => {
-      expect(sdk.naming.getCloudWatchLogLogicalId('functionName', 0)).to.equal(
-        'FunctionNameLogsSubscriptionFilterCloudWatchLog0'
+    it('should normalize the function name and add the standard suffix including the log group name', () => {
+      expect(sdk.naming.getCloudWatchLogLogicalId('functionName', '/aws/lambda/test')).to.equal(
+        'FunctionNameLogsSubscriptionFilterCloudWatchLogAwsLambdaTest'
```

Comment:
This could result in a breaking change because the logical id can be too long (that was the reason we've used integers in the past). I'm not sure what the limit for the length of the resource logical id is, but it might make sense to check that we're compliant with that.